app/waf/src/models/application.model.ts

```diff
@@ -12,13 +12,82 @@ export class Application extends CommonEntity {
   })
   status?: string;
 
+  @property({
+    type: 'string',
+    required: false,
+    schema: {
+      create: true,
+      update: true,
+      response: true,
+      example: '11111111-2222-3333-4444-555555555555',
+    },
+  })
+  adcId?: string;
+
+  @property({
+    type: 'string',
+    required: false,
+    schema: {
+      update: true,
+      response: true,
+      example: '11111111-2222-3333-4444-555555555555',
+    },
+  })
+  currentDeclarationId?: string;
```

Comment:
It will produce data redundancy. you have to handle the setting/maintain on the value for application changes or new declaration.
In fact, this information can be gotten via querying declaration.

Reply:
It is not redundant data. If someone hopes to deploy the config of one application without inputting a specific declaration id, this property is required. Otherwise, everyone need to know exactly which declaration is the right one and which is not.

We can let our app not to automatically specify this id. Just let the user to specify it by himself.

Reply:
I think requiring the user to generate the declaration before deploying it will make the workflow clear. 
That means `deploy` action belongs to declaration. Not application.

Reply:
Both of them should have deploy action. In this change set, we only implement one of them.

Reply:
Still, feel that letting `application` to have only the configuration API set and move deploying operations to `declaration` makes the frame simpler. 
Up to you. Got your meaning.